addons/editor-dark-mode/text_color.js

```diff
@@ -0,0 +1,110 @@
+function parseHex(hex) {
+  return {
+    r: parseInt(hex.substring(1, 3), 16),
+    g: parseInt(hex.substring(3, 5), 16),
+    b: parseInt(hex.substring(5, 7), 16),
+  };
+}
+
+function setCSSVar(name, value) {
+  document.documentElement.style.setProperty(`--editorDarkMode-${name}`, value);
+}
+
+function textColor(varName, hex, black, white, threshold) {
+  const { r, g, b } = parseHex(hex);
+  threshold = threshold !== undefined ? threshold : 170;
+  if (r * 0.299 + g * 0.587 + b * 0.114 > threshold) {
+    // https://stackoverflow.com/a/3943023
+    setCSSVar(varName, black !== undefined ? black : "#575e75");
+  } else {
+    setCSSVar(varName, white !== undefined ? white : "#ffffff");
+  }
+}
+
+function transparentVariant(varName, hex, opacity) {
+  const { r, g, b } = parseHex(hex);
+  setCSSVar(varName, `rgba(${r}, ${g}, ${b}, ${opacity})`);
+}
+
+function lightDarkVariant(varName, hex, threshold, cr, cg, cb, lr, lg, lb) {
+  const { r, g, b } = parseHex(hex);
+  if (lr === undefined) lr = cr;
+  if (lg === undefined) lg = cg;
+  if (lb === undefined) lb = cb;
+  textColor(
+    varName,
+    hex,
+    `rgb(${cr * r}, ${cg * g}, ${cb * b})`,
+    `rgb(${(1 - lr) * 255 + lr * r}, ${(1 - lg) * 255 + lg * g}, ${(1 - lb) * 255 + lb * b})`,
+    threshold
+  );
+}
+
+function testAll(settings) {
+  for (let setting of [
+    "page",
+    "primary",
+    "menuBar",
+    "selector",
+    "selector2",
+    "selectorSelection",
+    "accent",
+    "input",
+    "categoryMenu",
+    "palette",
+  ]) {
+    textColor(`${setting}-text`, settings.get(setting));
+  }
+  textColor("page-tabHoverFilter", settings.get("page"), "grayscale(100%)", "brightness(0) invert(1)");
+  textColor("primary-filter", settings.get("primary"), "brightness(0.4)", "none");
+  textColor("primary-filter2", settings.get("primary"), "none", "brightness(0) invert(1)");
+  textColor("secondary-filter", settings.get("secondary"), "brightness(0.4)", "none");
```

Comment:
This line isn't used for anything, and throws an error because the "secondary" setting doesn't exist. It should be removed - this fixes https://github.com/ScratchAddons/ScratchAddons/pull/1805#discussion_r593939501!